storagenode/storagenodedb/infodb.go

```diff
@@ -0,0 +1,163 @@
+	mu sync.Mutex
+	db *sql.DB
+}
+
+func NewInfo(path string) (*infodb, error) {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
+		return nil, err
+	}
+
+	db, err := sql.Open("sqlite3", "file:"+path+"?_journal=WAL")
+	if err != nil {
+		return nil, ErrInfo.Wrap(err)
+	}
+
+	return &infodb{db: db}, nil
+}
+
+func NewInfoInMemory() (*infodb, error) {
+	db, err := sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		return nil, ErrInfo.Wrap(err)
+	}
+
+	return &infodb{db: db}, nil
+}
+
+// Close closes any resources.
+func (db *infodb) Close() error {
+	return db.db.Close()
+}
+
+func (db *infodb) locked() func() {
+	db.mu.Lock()
+	return db.mu.Unlock
+}
+
+// CreateTables creates any necessary tables.
+func (db *infodb) CreateTables(log *zap.Logger) error {
+	migration := db.Migration()
+	return migration.Run(log.Named("migration"), db)
+}
+
+// RawDB returns access to the raw database, only for migration tests.
+func (db *infodb) RawDB() *sql.DB { return db.db }
+
+// Begin begins transaction
+func (db *infodb) Begin() (*sql.Tx, error) { return db.db.Begin() }
+
+// Rebind rebind parameters
+func (db *infodb) Rebind(s string) string { return s }
+
+// Schema returns schema
+func (db *infodb) Schema() string { return "" }
+
+// Migration returns table migrations.
+func (db *infodb) Migration() *migrate.Migration {
+	return &migrate.Migration{
+		Table: "versions",
+		Steps: []*migrate.Step{
+			{
+				Description: "Initial setup",
+				Version:     0,
+				Action: migrate.SQL{
+					// table for keeping serials that need to be verified against
+					`CREATE TABLE used_serial (
+						satellite_id  BLOB NOT NULL,
+						serial_number BLOB NOT NULL,
+						expiration    TIMESTAMP NOT NULL
+					)`,
+					// primary key on satellite id and serial number
+					`CREATE UNIQUE INDEX pk_used_serial ON used_serial(satellite_id, serial_number)`,
+					// expiration index to allow fast deletion
+					`CREATE INDEX idx_used_serial ON used_serial(expiration)`,
+
+					// certificate table for storing uplink/satellite certificates
+					`CREATE TABLE certificate (
+						cert_id       INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+						node_id       BLOB        NOT NULL,
+						peer_identity BLOB UNIQUE NOT NULL -- PEM encoded
+					)`,
+
+					// table for storing piece meta info
+					`CREATE TABLE pieceinfo (
+						satellite_id     BLOB      NOT NULL,
+						piece_id         BLOB      NOT NULL,
+						piece_size       BIGINT    NOT NULL,
+						piece_expiration TIMESTAMP NOT NULL, -- date when it can be deleted
+
+						uplink_piece_hash BLOB    NOT NULL, -- serialized pb.PieceHash signed by uplink
+						uplink_cert_id    INTEGER NOT NULL,
+
+						FOREIGN KEY(uplink_cert_id) REFERENCES certificate(cert_id)
+					)`,
+					// primary key by satellite id and piece id
+					`CREATE UNIQUE INDEX pk_pieceinfo ON pieceinfo(satellite_id, piece_id)`,
+
+					// table for storing bandwidth usage
+					`CREATE TABLE bandwidth_usage (
```

Comment:
Is that table there for keeping track of the amount transfered for the Dashboard?
If so, are we resetting that table every month?

Reply:
at the moment there's no resetting code, because I'm unsure what the exact logic should be

Reply:
The usage monitor should reset every 1st of the month.
Thats how our bandwidth metric has been published to work.

Reply:
we can query a particular month or rolling month whichever is necessary

Reply:
with regards to resetting things we can add that later as it's not critical part of the system

Reply:
Sounds good to me! Nice work